### DISCUSSION TITLE
USBを接続したときの挙動について

### DISCUSSION THREAD

**Discussion by hiroki-tagami** [Yowkees/keyball#470] [Q&A]
すべて組み立てが完了（LEDは未実装）し、すべてのキーが動作することも確認できて快適に使えているのですが
PCへの接続時、必ずTRRSケーブルを接続しなおさないと使用を開始することができません。
念のためTRRSケーブルも交換してみましたが、挙動は変わりませんでした。

使用が開始できるようになるまでは具体的には以下のような流れになります
（左右どちらにUSBを挿した場合でも同じ挙動になるようです）

0. 左右のTRRSケーブルを挿した状態でUSB-CをProMicroに接続
1. 左右のLDCに`keyball`の表示が出るが、キー入力などの操作はできない
2. USBを接続している方のTRRSケーブルを接続し直す
3. USBを挿した側のLCDの表示がKeyなどのステータスが見えているものに変わる  

この流れでの接続後は問題なく利用できているのですが
接続し直すたびにTRRSケーブルを挿しなおす必要があるものなのでしょうか？　
もしチェックするべき箇所などがあるようでしたら教えていただけると嬉しいです。
よろしくお願いします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

USBケーブルを挿したまま（通電状態）でTRSケーブルを抜き差しするとProMicroが壊れる可能性があります。
左右両方のOLEDにKeyballロゴが表示されるのは、PCが正しくUSBケーブルと認識できなかった状態です。
これはProMicroのUSB-typeCコネクタのクセみたいな物なのですが、ProMicroにUSBケーブルを”素早く””奥まで”差し込むと認識しやすいです。
両手にKeyballロゴが表示された場合は上記のようにUSBケーブルの差し直しでご対応いただければ幸いです。

**Comment by hiroki-tagami** [comment]
ご確認ありがとうございます。

ProMicroのUSB-Cの仕様ということで承知しました。

> ProMicroにUSBケーブルを”素早く””奥まで”差し込むと認識しやすいです。

こちらを試してみるとおっしゃる通り意識して素早く奥まで差し込むことで認識されることを確認できました。
（これまでもごく稀にTRRSケーブルを抜き差しせずともたまに認識することがあったので実装上の問題かと疑っていました）

解決しましたのでクローズさせていただきます、ありがとうございました
